File: csharp/main/Decorator.cs
using System.Collections.Generic;
using JetBrains.Annotations;

namespace Deltix.Luminary
{
	public sealed class Decorator
	{
		public Decorator([NotNull] DecoratorDef definition, [CanBeNull] List<DecoratorPropertyValue> arguments)
		{
			Definition = definition;
			Arguments = arguments ?? new List<DecoratorPropertyValue>();
		}

		[NotNull]
		public DecoratorDef Definition { get; }

		[NotNull, ItemNotNull]
		public List<DecoratorPropertyValue> Arguments { get; }
	}
}

File: java/src/main/java/deltix/luminary/Decorator.java
package deltix.luminary;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

public class Decorator {
    private final DecoratorDef definition;
    private final List<DecoratorPropertyValue> arguments;

    public Decorator(DecoratorDef definition, List<DecoratorPropertyValue> arguments) {
        this.definition = definition;
        this.arguments = arguments != null ? arguments : new ArrayList<DecoratorPropertyValue>();
    }

    @NotNull
    public DecoratorDef getDefinition() {
        return definition;
    }

    @NotNull
    public List<DecoratorPropertyValue> getArguments() {
        return arguments;
    }
}
